search/eval/eval.md

```diff
@@ -1,46 +1,198 @@
-Note the use of the `setEnum()` function. It allows you to set an enumerated value which will be passed downstream. The enumerated value doesn’t have to be a string; it can be a boolean, a number, a slice, or even an IP.
+#### Punctuation
 
-The eval module supports `if` and `switch` logic statements, as demonstrated below:
+Eval uses punctuation similar to the C programming language. For example, `( )` are used to enclose expressions in statements as in `if (foo == "bar")...`.
+
+The following character sequences are reserved punctuation.
 
 ```
-if len(Body) <= 10 { setEnum("postlen", "short"); setEnum(“anotherEnum”, “foo”) }
+( )		Expression and function grouping
+{ }		Statement and block grouping
+,		Function parameter separator
+;		Statement separator
+$		Alternative enumerated value accessor
+[ ]		Array indexing (not currently supported)
+. :		Scope and range (not currently supported)
 ```
 
+#### Numbers
+
+Numbers can be expressed as decimal and hexadecimal integers, as well as decimal floating-point. Hexadecimal values always begin with "0x". 
+
+The following examples are valid numbers.
+
 ```
-switch DstPort { case 80: setEnum(“protocol”, “http”); case 22: setEnum(“protocol”, “ssh”); default: setEnum(“protocol”, “unknown”) }
+5
+-5
+0x05
+5.00
 ```
 
-### Further reference
+#### String literals
+
+Strings are any sequence of characters enclosed in double quotes `"`. 
+
+### Constants
+
+Constants consist of any numbers or string literals, as well as a special case of an identifier that otherwise is not a declared variable or enumerated value from earlier in the query.
+
+### Variables and types
+
+All enumerated values created in the query pipeline are available in eval as variables. Additionally, all variables created in eval are available as enumerated values to any modules after eval in the query pipeline. 
+
+Since enumerated values can have a different type from entry to entry, eval may behave differently based on the type of a variable at runtime. For example, the filter `foo < 5` will only work if "foo" can be interpreted at runtime as a number. It is not an error for "foo" to be a different type (in this example the boolean expression "foo < 5" would be false if foo is not a number). 
+
+The `type()` built-in function can be used at runtime to determine the type of a variable in a given entry.
 
-* [The Gravwell documentation for the Anko language](/scripting/scripting) is a generic description of the Anko scripting language
-* [The eval module article](/scripting/eval) describes the eval module in more detail.
+`eval` can attempt to cast a variable to another type using the following cast operators.
+
+```
+int
+float
+string
+ip
+mac
+duration
+time
+type
+```
+
+Additionally, eval will attempt to "promote" types that can be automatically cast. For example, if "foo" contains the string "3.14", the expression `foo < 3.14` will cause eval to attempt to promote foo to a floating-point number before the expression is evaluated.
```

Comment:
can you change one of the "3.14" occurrences to something else so it's more clear which is the contents of the variable and which is the constant being compared against?